Sources/AST/ASTPassContext.swift

```diff
@@ -112,6 +112,12 @@ extension ASTPassContext {
     return functionDeclarationContext != nil || initializerDeclarationContext != nil
   }
 
+  // Whether we are visiting a node inside the rhs of an assignment.
+  public var inAssignment: Bool {
+    get { return self[IsAssignment.self] ?? false }
+    set { self[IsAssignment.self] = newValue }
```

Comment:
I found what I was vaguely referring to last review: "asLValue" is set every time we are visiting the lhs of a boolean operator, can I recommend we expand this to an enum i.e. "side" with cases Left and Right (which would be my preference), or make "asLValue" consistent with this or vice versa

Reply:
Can we not combine the two: so we check lValue/rValue and a check for the op being assignment?

Reply:
OK, so inAssignment would be true for both the LHS and RHS of an assignment. I can't see an immediate use for that but if you think it would be useful I will change it.

Reply:
It is in an assignment so it should be different.